### ISSUE TITLE
Cannot stringify arbitrary non-POJOs GraphModel Warning

### ISSUE THREAD

**Issue by lechim** [Gomah/nuxt-shopify#182]
Keep getting Cannot stringify arbitrary non-POJOs GraphModel warning

In store/index.js
`async nuxtServerInit({commit, dispatch}, context) {`
`        if (process.server) {`
`            try{`
`                const products = await context.app.$shopify.product.fetchAll();`
`                commit('products/setProducts', products)`
`            }catch (e) {`
`                console.log('products err: ',e)`
`            }`
`        }`
`    }`

I get same error in asyncData.

Is there something I am missing?

**Comment by Gomah** [comment]
Correct, Nuxt uses [devalue](https://github.com/nuxt/devalue) to [stringify](https://github.com/nuxt/devalue/blob/master/src/index.ts#L80) Shopify's [GraphModel](https://shopify.github.io/graphql-js-client/graph-model.js.html).

**Comment by doomb0t** [comment]
Howdy I am open to taking a stab at this as I am also having this issue on a side project I am working on. The plan is to leverage nuxt-shopify and vuetify to provide an open source store front that can be hosted on github/lab pages so folks only have to pay for the storefront api access.

Anyhow, I will start on this asap

**Comment by ideacco** [comment]
I also encountered the same problem and hope to fix it soon. Thank you

**Comment by prisme** [comment]
Hi, I'm also getting this warning when running `nuxt generate`, and I suspect it is what's causing my Netlify deploy to static target to fail 404 on the / route 

Here's my early stage repo where I basically only installed nuxt, nuxt-shopify and prismic :
https://github.com/prisme/jeannesemarie deployed to https://jeannesemarie.netlify.app/

On local dev env, `nuxt generate` issues warnings but renders index page as expected: shopify products are rendered in dist/index.html
On Netlify, `nuxt generate` issues an undefined error and I get 404 for index page. Unless I skip the call to `$shopify.fetchAll()` in pages/index.vue

Hoping to provide a reduced case insight on this issue, and maybe find help in solving my deploy problem :)

Netlify log
```
$ npm run generate
> jeannesemarie@1.0.0 generate /opt/build/repo
> nuxt generate
[error]  /
 
[object Object]
[error] [ { message: 'an unknown error has occurred.' } ]
```

**Comment by juanpprieto** [comment]
+1

**Comment by binoysarker** [comment]
i am also having this problem

**Comment by adamdehaven** [comment]
@Gomah it looks like another project resolved this already; I'm guessing you could do something similar. [Here's the link for their PR](https://github.com/vuex-orm/vuex-orm/pull/283/commits/b304cdc7599ee297b96996d5201c7ec094139e1f)

**Comment by rcobiella** [comment]
Any updates on this? Thank you!

**Comment by sdevogel** [comment]
+1

**Comment by SantosJMM** [comment]
I have the same, I hope they can solve it soon, thank you very much

**Comment by riccardolardi** [comment]
+1

**Comment by Asingerdev** [comment]
Any update on this? Starting a new site and was looking to use this package

**Comment by dougwl** [comment]
Any news?

**Comment by fmoessle** [comment]
Having this problem too. Using nuxt3 with useAsyncData.
